packages/apis/uniswapv2/src/__tests__/e2e/fetch_e2e.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { buildAndDeployApi, initTestEnvironment, stopTestEnvironment } from "@web3api/test-env-js";
+import { UriRedirect, Web3ApiClient } from "@web3api/client-js";
+import { ChainId, Pair, Token, TokenAmount } from "./types";
+import path from "path";
+import { defaultUniswapTokenList, getRedirects, getTokenList } from "../testUtils";
+
+jest.setTimeout(60000);
+
+// mainnet-fork:
+// image: 'trufflesuite/ganache-cli'
+// ports:
+//   - '8546:8545'
+// command: -l 8000000 --deterministic --hostname=0.0.0.0 --chainId 1 --fork https://mainnet.infura.io/v3/d119148113c047ca90f0311ed729c466
+
+describe("Fetch", () => {
+
+  let client: Web3ApiClient;
+  let ensUri: string;
+  let tokens: Token[] = [];
+
+  beforeAll(async () => {
+    const { ethereum: testEnvEtherem, ensAddress, ipfs } = await initTestEnvironment();
+    // get client
+    const redirects: UriRedirect[] = getRedirects(testEnvEtherem, ipfs, ensAddress);
+    client = new Web3ApiClient({ redirects });
+    // deploy api
+    const apiPath: string = path.resolve(__dirname + "/../../../");
+    const api = await buildAndDeployApi(apiPath, ipfs, ensAddress);
+    ensUri = `ens/testnet/${api.ensDomain}`;
+    // ipfsUri = `ipfs/${api.ipfsCid}`;
+    // set up test case data
+    tokens = await getTokenList(defaultUniswapTokenList);
+  });
+
+  afterAll(async () => {
+    await stopTestEnvironment();
+  });
+
+  it("Fetches token data", async () => {
+    const tokenData = await client.query<{
+      fetchTokenData: Token;
+    }>({
+      uri: ensUri,
+      query: `
+        query {
+          fetchTokenData(
+            chainId: $chainId
+            address: $address
+          )
+        }
+      `,
+      variables: {
+        chainId: tokens[0].chainId,
+        address: tokens[0].address,
+      },
+    });
+
+    expect(tokenData.errors).toBeFalsy();
```

Comment:
Is there any way to assert the value in this tests?
This is not really that usefull. We should probably call 
the ether separately for the result and compare.

Reply:
Still working on this. Will update you when it's done.

Reply:
done